## 75% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 1
Expected to fail tests: 1
Passed tests: 6
```
```
IDs of failed tests: 703133 
```
```
IDs of expected to fail tests: 702843 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/network-drawer-back&PR_ID=21689&APK_NAME=21689.apk&TR_CASE_IDS=703133">Rerun failed tests</a></li><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3736074">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div># STEP: Check that removed user is not shown in the list anymore</div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:240: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:192: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 zQ3...dWXh5 was not restored as a contact from waku backup!
E    zQ3...Vacac was not restored as a contact from waku backup!
E    admin_open was not restored from waku-backup!!
E    member_open was not restored from waku-backup!!
E    admin_closed was not restored from waku-backup!!
E    member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7c261df026904a78b10662b8fb8a4aa7?auth=9549ec43099335e08ad1fe920221088d#78">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3736073">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Skipped due to waku issue on staging fleet]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (6)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3736077">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3736078">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3736072">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fea5f3af9c8d493fa3d886761b7cf650?auth=cccb737812cc1a347699cdd6f38f82c4#107">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/55f01414bdf4445eb6a14afac3609057?auth=872cc443fae9865049d9ceab5595865e#94">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3736076">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3736075">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3736071">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7c261df026904a78b10662b8fb8a4aa7?auth=9549ec43099335e08ad1fe920221088d#46">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/wallet/swap/events.cljs
+++ b/src/status_im/contexts/wallet/swap/events.cljs
@@ -16,7 +16,9 @@
          test-networks-enabled? (get-in db [:profile/profile :test-networks-enabled?])
          view-id                (:view-id db)
          root-screen?           (or (= view-id :wallet-stack) (nil? view-id))
-         account                (or from-account (swap-utils/wallet-account wallet))
+         available-accounts     (utils/get-accounts-with-token-balance (:accounts wallet)
+                                                                       (:asset-to-pay data))
+         account                (or from-account (swap-utils/wallet-account wallet available-accounts))
          asset-to-pay           (if (get-in data [:asset-to-pay :networks])
                                   (:asset-to-pay data)
                                   (swap-utils/select-asset-to-pay-by-symbol
@@ -24,8 +26,7 @@
                                     :account                account
                                     :test-networks-enabled? test-networks-enabled?
                                     :token-symbol           (get-in data [:asset-to-pay :symbol])}))
-         multi-account-balance? (-> (utils/get-accounts-with-token-balance (:accounts wallet)
-                                                                           asset-to-pay)
+         multi-account-balance? (-> available-accounts
                                     (count)
                                     (> 1))
          network'               (or network

--- a/src/status_im/contexts/wallet/swap/events.cljs
+++ b/src/status_im/contexts/wallet/swap/events.cljs
@@ -18,7 +18,9 @@
          root-screen?           (or (= view-id :wallet-stack) (nil? view-id))
          available-accounts     (utils/get-accounts-with-token-balance (:accounts wallet)
                                                                        (:asset-to-pay data))
-         account                (or from-account (swap-utils/wallet-account wallet available-accounts))
+         account                (or from-account
+                                    (swap-utils/current-viewing-account wallet)
+                                    (first available-accounts))
          asset-to-pay           (if (get-in data [:asset-to-pay :networks])
                                   (:asset-to-pay data)
                                   (swap-utils/select-asset-to-pay-by-symbol

--- a/src/status_im/contexts/wallet/swap/utils.cljs
+++ b/src/status_im/contexts/wallet/swap/utils.cljs
@@ -30,18 +30,13 @@
     :else
     (i18n/label :t/something-went-wrong-please-try-again-later)))
 
-(defn wallet-account
-  "Picks the account that's gonna be used for the swap operation.
-   It's gonna be either the preselected account defined by
-   `[:wallet :current-viewing-account-address]` in `db`
-   or the first operable account from the list of available accounts (accounts with token balance)."
-  [wallet available-accounts]
-  (if-let [wallet-address (get wallet :current-viewing-account-address)]
+(defn current-viewing-account
+  [wallet]
+  (when-let [wallet-address (get wallet :current-viewing-account-address)]
     (-> wallet
         :accounts
         vals
-        (utils/get-account-by-address wallet-address))
-    (first available-accounts)))
+        (utils/get-account-by-address wallet-address))))
 
 (defn select-asset-to-pay-by-symbol
   "Selects an asset to pay by token symbol.

--- a/src/status_im/contexts/wallet/swap/utils.cljs
+++ b/src/status_im/contexts/wallet/swap/utils.cljs
@@ -33,10 +33,7 @@
 (defn current-viewing-account
   [wallet]
   (when-let [wallet-address (get wallet :current-viewing-account-address)]
-    (-> wallet
-        :accounts
-        vals
-        (utils/get-account-by-address wallet-address))))
+    (get-in wallet [:accounts wallet-address])))
 
 (defn select-asset-to-pay-by-symbol
   "Selects an asset to pay by token symbol.

--- a/src/status_im/contexts/wallet/swap/utils.cljs
+++ b/src/status_im/contexts/wallet/swap/utils.cljs
@@ -30,33 +30,24 @@
     :else
     (i18n/label :t/something-went-wrong-please-try-again-later)))
 
-(defn- first-operable-account
-  [accounts]
-  (->> accounts
-       vals
-       (remove :watch-only?)
-       (filter :operable?)
-       (sort-by :position)
-       first))
-
 (defn wallet-account
-  "Picks the account that's gonna be used for the swap operation. 
-   It's gonna be either the preselected account defined by 
+  "Picks the account that's gonna be used for the swap operation.
+   It's gonna be either the preselected account defined by
    `[:wallet :current-viewing-account-address]` in `db`
-   or the first operable account."
-  [wallet]
+   or the first operable account from the list of available accounts (accounts with token balance)."
+  [wallet available-accounts]
   (if-let [wallet-address (get wallet :current-viewing-account-address)]
     (-> wallet
         :accounts
         vals
         (utils/get-account-by-address wallet-address))
-    (-> wallet :accounts first-operable-account)))
+    (first available-accounts)))
 
 (defn select-asset-to-pay-by-symbol
   "Selects an asset to pay by token symbol.
    It's used for cases when only token symbol is available and the information
-   about token needs to be extracted from the database. 
-   That happens when token is being selected on the home screen and 
+   about token needs to be extracted from the database.
+   That happens when token is being selected on the home screen and
    it basically indicates that no account pre-selection was made."
   [{:keys [wallet account test-networks-enabled? token-symbol]}]
   (let [networks (-> (get-in wallet [:networks (if test-networks-enabled? :test :prod)])
@@ -73,7 +64,7 @@
 
 (defn select-default-asset-to-receive
   "Selects an asset to receive if it was not provided explicitly.
-   The principle of how the asset is being selected is simple: we get the 
+   The principle of how the asset is being selected is simple: we get the
    whole list, remove the currently selected `asset-to-pay` from it, and choose
    the first one of what's left."
   [{:keys [wallet account test-networks-enabled? asset-to-pay]}]
@@ -86,7 +77,7 @@
 (defn select-network
   "Chooses the network.
    Usually user needs to do the selection first and if the selection was done
-   then the list of networks for the defined token will always contain 
+   then the list of networks for the defined token will always contain
    one entry. Otherwise `nil` will be returned from here which will serve
    as an indicator that the network selector needs to be displayed."
   [{:keys [networks]}]

